### DISCUSSION TITLE
Select-Component opens automatically in combination with Dropzone.

### DISCUSSION THREAD

**Discussion by michaelhuber1937** [mantinedev/mantine#1660] [Q&A] [answered]
I have an issue with the Select-Component.

I am using this component in a Modal alongside a Dropzone. When the user is selecting an item via the select-component, this component has still the focus. The dropdown is closed but the focus of Select is still active. I think this is intended behaviour.

When I then Drag & Drop a file into the Dropzone on the Modal-View, the Select opens automatically (which is horrible UX). This has obviously to do with the focus-state of the select-component. Because when I deselect the Select-component before Drag&Drop manually, everything is fine!

So is there a possible solution that the Select-Component does NOT open automatically when a user drag & drop a file? Like a flag "blurOnClose", so the Select-component automatically looses its focus-state when selection an item?

Comment: the Select opens in the moment the file hovers for the first over the viewport.

I already tried a lot of things, e.g. handling the onDropdownClose-event of Select-Component and tried to call the blur-function of the underlying input-field (via ref). This does not work.

Furthermore I tried to set the focus to another element, this also does not work. The select-component does not loose its focus-state.

Any ideas?

**Comment by rtivital** [comment, marked as answer]
You can move focus to a dropzone or other element when the user starts dragging a file over the viewport. This happens because when the user interacts with other application, the browser loses focus (blur event is triggered on Select component), then when the user drags the file over the viewport browser window become active, and it focuses the last active element.

**Reply by michaelhuber1937** [reply]
I tried your idea, thanks for that!

It works but in my opinion it is even worse UX. Because when I enter the viewport while dragging, the Select opens and immediately shuts down when calling the blur. This results in a flickering.

But you gave me an idea which works fine! When the user wants to drag and drop a file, one has to select the file in the OS. So that results in losing the focus of the entire browser-window. So I am listening to this event and call the blur on the Select-component.

Of course I detach the event-handler when the component is unmounted (clean-up, return-value is a function of useEffect).

```
const MySelect = ({ ...props }) => {
    const refSelect = useRef();

    const onBlur = () => {
        refSelect.current.blur();
    };

    useEffect(() => {
        window.addEventListener('blur', onBlur);

        return () => {
            window.removeEventListener('blur', onBlur);
        };
    }, []);

    return (
        <Select ref={refSelect} {...props} />
    );
};
```
